### PR TITLE
[SPARK-44299][SQL] Assign names to the error class _LEGACY_ERROR_TEMP_227[4-6,8]

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to assign names to the error class, include: 
- _LEGACY_ERROR_TEMP_2274 => UNSUPPORTED_FEATURE.REPLACE_NESTED_COLUMN
- _LEGACY_ERROR_TEMP_2275 => CANNOT_INVOKE_IN_TRANSFORMATIONS
- _LEGACY_ERROR_TEMP_2276 => UNSUPPORTED_FEATURE .HIVE_WITH_ANSI_INTERVALS
- _LEGACY_ERROR_TEMP_2278 => INVALID_FORMAT.MISMATCH_INPUT

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Update & Add new UT.
- Manually test.
- Pass GA.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @MaxGekk

File: common/utils/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -792,6 +797,11 @@
       "The expression <sqlExpr> cannot be used as a grouping expression because its data type <dataType> is not an orderable data type."
     ]
   },
+  "HIVE_WITH_ANSI_INTERVALS_IS_UNSUPPORTED" : {
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-classes.json:800]: How about a sub-class of `UNSUPPORTED_FEATURE`?

File: common/utils/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -1740,6 +1750,11 @@
     ],
     "sqlState" : "23K01"
   },
+  "MISMATCH_NUMBER_FORMAT" : {
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-classes.json:1753]: Maybe a sub-class of `INVALID_FORMAT` or re-use an existing one?

File: common/utils/src/main/resources/error/error-classes.json [hunk 3]
```diff
@@ -2114,6 +2129,11 @@
     ],
     "sqlState" : "42614"
   },
+  "REPLACE_NESTED_COLUMN_IS_UNSUPPORTED" : {
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-classes.json:2132]: Sub-class of `UNSUPPORTED_FEATURE`?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 4]
```diff
@@ -2592,14 +2592,14 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
 
   def nestedFieldUnsupportedError(colName: String): SparkUnsupportedOperationException = {
     new SparkUnsupportedOperationException(
-      errorClass = "_LEGACY_ERROR_TEMP_2274",
+      errorClass = "REPLACE_NESTED_COLUMN_IS_UNSUPPORTED",
       messageParameters = Map(
         "colName" -> colName))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:2597]:
```suggestion
      messageParameters = Map("colName" -> toSQLId(colName)))
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 5]
```diff
@@ -2671,9 +2671,10 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
         "functionName" -> toSQLId("aes_encrypt")))
   }
 
-  def hiveTableWithAnsiIntervalsError(tableName: String): SparkUnsupportedOperationException = {
+  def hiveTableWithAnsiIntervalsError(
+      tableName: String): SparkUnsupportedOperationException = {
     new SparkUnsupportedOperationException(
-      errorClass = "_LEGACY_ERROR_TEMP_2276",
+      errorClass = "HIVE_WITH_ANSI_INTERVALS_IS_UNSUPPORTED",
       messageParameters = Map("tableName" -> tableName))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:2678]: Please, quote by `toSQLId`

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: docs/sql-error-conditions-invalid-format-error-class.md [hunk 6]
```diff
@@ -49,6 +49,10 @@ The escape character is not allowed to end with.
 
 The escape character is not allowed to precede `<char>`.
 
+## MISMATCH_INPUT
+
+The input <inputType> '<input>' does not match the format.
```
Comment (MaxGekk) [inline docs/sql-error-conditions-invalid-format-error-class.md:54]: I think we should follow our convention and don't perform quoting in error formats.
```suggestion
The input <inputType> <input> does not match the format.
```

File: common/utils/src/main/resources/error/error-classes.json [hunk 7]
```diff
@@ -1186,6 +1191,11 @@
           "The escape character is not allowed to precede <char>."
         ]
       },
+      "MISMATCH_INPUT" : {
+        "message" : [
+          "The input <inputType> '<input>' does not match the format."
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-classes.json:1196]: nit:
```suggestion
          "The input <inputType> <input> does not match the format."
```

**Review by MaxGekk — APPROVED** [review]
LGTM, let's address the minor comment in your another PR.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

File: docs/sql-error-conditions-invalid-format-error-class.md (hunk 6 above)
Comment (panbingkun) [reply docs/sql-error-conditions-invalid-format-error-class.md:54]: Okay.

**Pull request closed without merging**